Formula/samba.rb

```diff
@@ -0,0 +1,123 @@
+        system "perl", "Makefile.PL", "INSTALL_BASE=#{libexec}"
+        system "make"
+        system "make", "install"
+      end
+    end
+    # CFLAGS is for avoiding hitting https://bugzilla.samba.org/show_bug.cgi?id=14680 .
+    # Remove this CFLAGS when the patch ( https://attachments.samba.org/attachment.cgi?id=16579 ) gets merged.
+    ENV.append "CFLAGS", "-include #{buildpath}/lib/util/debug.h"
+    system "./configure",
+           "--disable-cephfs",
+           "--disable-cups",
+           "--disable-iprint",
+           "--disable-glusterfs",
+           "--disable-python",
+           "--without-acl-support",
+           "--without-ad-dc",
+           "--without-ads",
+           "--without-dnsupdate",
+           "--without-ldap",
+           "--without-libarchive",
+           "--without-json",
+           "--without-ntvfs-fileserver",
+           "--without-pam",
+           "--without-regedit",
+           "--without-syslog",
+           "--without-utmp",
+           "--without-winbind",
+           "--prefix=#{prefix}"
+    system "make"
+    system "make", "install"
+    on_macos do
+      # macOS has its own SMB daemon as /usr/sbin/smbd, so rename our smbd to samba-dot-org-smbd to avoid conflict.
+      # samba-dot-org-smbd is used by qemu.rb .
+      mv "#{sbin}/smbd", "#{sbin}/samba-dot-org-smbd"
+    end
+  end
+
+  def caveats
+    on_macos do
+      <<~EOS
+        macOS has its own SMB daemon as /usr/sbin/smbd, so Samba version of smbd is installed as "#{HOMEBREW_PREFIX}/sbin/samba-dot-org-smbd"
+
+        On macOS, Samba should be executed as a non-root user: https://bugzilla.samba.org/show_bug.cgi?id=8773
+      EOS
+    end
+  end
+
+  test do
+    smbd = "#{sbin}/smbd"
+    on_macos do
+      smbd = "#{sbin}/samba-dot-org-smbd"
+    end
+
+    system smbd, "--build-options"
+    system smbd, "--version"
+
+    mkdir_p "samba/state"
+    mkdir_p "samba/data"
+    (testpath/"samba/data/hello").write "hello"
+
+    # mimic smb.conf generated by qemu
+    # https://github.com/qemu/qemu/blob/v6.0.0/net/slirp.c#L862
+    (testpath/"smb.conf").write <<~EOS
+      [global]
+      private dir=#{testpath}/samba/state
+      interfaces=127.0.0.1
+      bind interfaces only=yes
+      pid directory=#{testpath}/samba/state
+      lock directory=#{testpath}/samba/state
+      state directory=#{testpath}/samba/state
+      cache directory=#{testpath}/samba/state
+      ncalrpc dir=#{testpath}/samba/state/ncalrpc
+      log file=#{testpath}/samba/state/log.smbd
+      smb passwd file=#{testpath}/samba/state/smbpasswd
+      security = user
+      map to guest = Bad User
+      load printers = no
+      printing = bsd
+      disable spoolss = yes
+      usershare max shares = 0
+      [test]
+      path=#{testpath}/samba/data
+      read only=no
+      guest ok=yes
+      force user=#{ENV["USER"]}
+    EOS
+
+    port = free_port
+    fork { exec smbd, "-S", "-F", "--configfile=smb.conf", "--port=#{port}", "--debuglevel=10" }
```

Comment:
`brew test` passes locally on my macOS 11 laptop but CI failing with `Server exit (EOF on stdin)` 

Maybe we need to supply custom stdin FD when TTY is detached?

Reply:
Possibly. You can use `spawn` to easily redirect stdin. See `neovim-remote` for an example.

Reply:
Added `spawn`, thanks